core/src/Pos/Core/NetworkMagic.hs

```diff
@@ -0,0 +1,47 @@
+module Pos.Core.NetworkMagic
+       ( NetworkMagic (..)
+       , makeNetworkMagic
+       ) where
+
+import           Universum
+
+import           Data.SafeCopy (SafeCopy (..), contain, safeGet, safePut)
+import           Data.Serialize (getWord8, putWord8)
+import qualified Data.Text.Buildable as Buildable
+import           Formatting (bprint, build, (%))
+
+import           Pos.Crypto.Configuration (ProtocolMagic (..), RequiresNetworkMagic (..),
+                                           getProtocolMagic)
+import           Pos.Util.Util (cerealError)
+
+
+--------------------------------------------------------------------------------
+-- NetworkMagic
+--------------------------------------------------------------------------------
+
+-- mhueschen: although it makes no sense for an identifier to have a sign, we're
+-- opting to maintain consistency with `ProtocolMagicId`, rather than risk subtle
+-- conversion bugs.
```

Comment:
We should be as correct as possible and provide small locally safe functions for conversion. In this case, providing `NMJust !Word32` would be Right, and we could have a function `ProtocolMagicId -> NetworkMagic` that handles it correctly.

Alternatively, a smart constructor like `x :: Int32 -> Maybe NetworkMagic` that fails on < 0.

Reply:
I believe we want negative values. The type of `ProtocolMagic`'s identifier should really be a `Word32`, but for historical reasons is an `Int32`. The point of the identifier is to disambiguate networks, so negative values also serve that purpose.